crates/evm/evm/src/executors/invariant/result.rs

```diff
@@ -155,3 +147,45 @@ pub(crate) fn can_continue(
     }
     Ok(RichInvariantResults::new(true, call_results))
 }
+
+/// Given the executor state, asserts conditions within `tearDown` function.
+/// If call fails then the invariant test is considered failed.
+pub(crate) fn assert_tear_down(
+    invariant_contract: &InvariantContract<'_>,
+    invariant_config: &InvariantConfig,
+    targeted_contracts: &FuzzRunIdentifiedContracts,
+    executor: &mut Executor,
```

Comment:
same q,

but perhaps we keep it in variant because these functions are only useful in here?

Reply:
Yeah, they're invariant related only and cannot be reused for other types of test, so probably better to keep them here